digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -326,77 +328,104 @@ pub async fn get_related_for_assets(
     if let Some(required) = required_creator {
         assets_map.retain(|_id, asset| asset.creators.iter().any(|c| c.creator == required));
     }
-
     let ids = assets_map.keys().cloned().collect::<Vec<_>>();
-    let authorities = asset_authority::Entity::find()
-        .filter(asset_authority::Column::AssetId.is_in(ids.clone()))
-        .order_by_asc(asset_authority::Column::AssetId)
-        .all(conn)
-        .await?;
+
+    // Fetch asset_data, creators, authorities, and other related data in parallel
+    let (authorities, tokens, attachments, groups) = tokio::join!(
+        // Fetch authorities
+        asset_authority::Entity::find()
+            .filter(asset_authority::Column::AssetId.is_in(ids.clone()))
+            .order_by_asc(asset_authority::Column::AssetId)
+            .all(conn),
+        // Fetch tokens (fungible assets)
+        async {
+            if options.show_fungible {
+                find_tokens(conn, ids.clone()).await
+            } else {
+                Ok(Vec::new())
+            }
+        },
+        // Fetch attachments (inscriptions)
+        async {
+            if options.show_inscription {
+                asset_v1_account_attachments::Entity::find()
+                    .filter(
+                        asset_v1_account_attachments::Column::AttachmentType
+                            .eq(V1AccountAttachments::TokenInscription),
+                    )
+                    .filter(asset_v1_account_attachments::Column::AssetId.is_in(ids.clone()))
+                    .all(conn)
+                    .await
+            } else {
+                Ok(Vec::new())
+            }
+        },
+        // Fetch groups
+        async {
+            let cond = if options.show_unverified_collections {
+                Condition::all()
+            } else {
+                Condition::any()
+                    .add(asset_grouping::Column::Verified.eq(true))
```

Comment:
Collection offers some constructs like mapping that I think we can use clean up the conditional adding of queries into the list.